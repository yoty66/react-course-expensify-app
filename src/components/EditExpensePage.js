import React from "react";
import { connect } from  'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense} from '../actions/expenses'

export class EditExpensePage extends React.Component{

    editExpense=(expense)=>{
        this.props.editExpense(this.props.expense.id ,expense);
        this.props.history.push('/')
    };

    removeExpense=(expense)=>{
      this.props.removeExpense(expense) ;
      this.props.history.push('/')
    };

    render=()=>{

        return(
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.editExpense}/>
                <button onClick={()=>{this.removeExpense(this.props.expense) }}>  Remove </button>
            </div>
        );
    }
};

const mapStateToProps = (state , props) =>{
    return {
        expense: state.expenses.find(({id}) => id === props.match.params.id )
    }
};

const mapDispatchToProps = ( dispatch ) => ({
   editExpense: (id ,expense) => dispatch(startEditExpense( id , expense) ),
   removeExpense: ( expense ) =>{
       const action =startRemoveExpense({ id: expense.id });
       dispatch(action) }
});
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);