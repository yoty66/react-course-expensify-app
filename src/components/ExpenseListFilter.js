import React from 'react';
import { connect } from 'react-redux';
import {sortByDate , sortByAmount, setEndDate, setStartDate } from '../actions/filters';
import {DateRangePicker, SingleDatePicker} from 'react-dates';

export class ExpenseListFilter extends React.Component {
    state = {
        calendarFocused: null
    };

    onDatesChange = ({startDate,endDate})=> {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };

    onFocusChange = (calendarFocus)=> {
        this.setState(()=> ({calendarFocused: calendarFocus}));
    };

    onTextChange = (e) => {
        this.props.setFilterText(e.target.value)
    };

    onSortChange = (e)=> {e.target.value === 'amount' ?
        this.props.sortByAmount() :
        this.props.sortByDate() };

    render(){
        return(
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input type={"text"} value={this.props.filters.text}
                              className={"text-input"}
                               placeholder={"Search expenses"}
                               onChange={this.onTextChange}/>
                    </div>

                    <div className="input-group__item">
                        <select value={this.props.filters.sortBy}
                                onChange={this.onSortChange}
                        className={"select"}>
                            <option value='date'>Date</option>
                            <option value='amount'>Amount</option>
                        </select>
                    </div>

                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calendarFocused}
                            onFocusChange={this.onFocusChange}
                            showClearDates={true}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                        />
                    </div>
                </div>
        </div>);
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };

};
const mapDispatchToProps = (dispatch) =>({
        sortByAmount: () => dispatch(sortByAmount()),
        sortByDate: () => dispatch(sortByDate()),
        setFilterText: (text) => dispatch({type: 'SET_FILTER_TEXT', text}),
        setStartDate :(startDate) => dispatch(setStartDate(startDate)),
        setEndDate :(endDate) => dispatch(setStartDate(endDate))
});

export default  connect(mapStateToProps , mapDispatchToProps)(ExpenseListFilter);