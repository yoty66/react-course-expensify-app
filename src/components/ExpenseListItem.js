import React from 'react';
import {NavLink} from "react-router-dom";


const ExpenseListItem = ({id,description, amount , createdAt }) => (
    <div>
        <NavLink to={`/edit/${id}`} > <h3>{description}</h3></NavLink>
        <p>{amount} -  {createdAt}</p>

    </div>
);

export default ExpenseListItem;