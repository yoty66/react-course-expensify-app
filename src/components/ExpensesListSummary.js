import React from 'react';
import numeral from 'numeral';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { getVisableExpenses } from '../selctors/expenses';
import getExpensesTotal from '../selctors/expenses-total'
export const ExpensesListSummary= ({expensesCount,expensesTotal})=>(
    <div className="page-header">
        <div className="content-container">
        <h1 className="page-header__title">
            Viewing <span>{expensesCount}</span> {expensesCount==1 ? 'expense ' :'expenses '} totaling <span>{numeral(expensesTotal/100).format('$0,0.00').replace('$','â‚ª')}</span>
        </h1>
        </div>
        <div className="page-header__actions">
            <Link className="button" to={'/create'}>
                Add Expense
            </Link>
        </div>
    </div>
        );

const mapStateToProps = (state)=> {
        const visibleExpenses = getVisableExpenses(state.expenses, state.filters);
        return{
            expensesCount:visibleExpenses.length, 
            expensesTotal: getExpensesTotal(visibleExpenses)
        };
};
export default connect(mapStateToProps)(ExpensesListSummary);


