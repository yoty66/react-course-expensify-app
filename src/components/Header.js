import { Link } from "react-router-dom";
import React from "react";
import { startLogout } from '../actions/auth';
import { connect } from 'react-redux';


export const Header = ({ startLogout }) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link to={'/dashboard'} className="header__title"  exact={true}>
                    <h1>Expensify</h1>
                </Link>
                <button onClick={startLogout} className="button button--link"> Logout </button>
             </div>
        </div>
    </header>
);

const mapDispatchToProps = ( dispatch ) =>({
  startLogout: ()=> dispatch(startLogout())
});

export default connect(null, mapDispatchToProps)(Header);