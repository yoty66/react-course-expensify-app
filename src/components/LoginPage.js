import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';


   const LoginPage = ({ startLogin })=>( <button onClick={startLogin}> Login </button>);


const mapDispatchToProps = ( dispatch )=> ({
   startLogin: ()=> dispatch(startLogin())
   // startLogin: ()=> console.log('123')
});

   export default connect(null , mapDispatchToProps)(LoginPage)




