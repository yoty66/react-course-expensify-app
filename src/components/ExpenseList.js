import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem'
import {getVisableExpenses} from '../selctors/expenses';

export const ExpenseList = (props) =>
    (
      <div>
          {
              props.expenses.length === 0
                  ? (<p>there are no items</p>)
                  : (props.expenses.map((expense)=> <ExpenseListItem {...expense}/>))
          }
      </div>
    );

const mapStateToProps = (state)=> ({
    expenses: getVisableExpenses(state.expenses ,state.filters)
});

export default connect(mapStateToProps)(ExpenseList);

