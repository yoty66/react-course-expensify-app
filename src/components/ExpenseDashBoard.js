import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilter';
import ExpenseListSummary from './ExpensesListSummary';

export default   ()=> (
    <div>
        <ExpenseListSummary/>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
);
