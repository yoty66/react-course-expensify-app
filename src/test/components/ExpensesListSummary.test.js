import  React from 'react';
import { ExpensesListSummary } from '../../../src/components/ExpensesListSummary'
import { shallow } from 'enzyme';

let wrapper;
test('should display two expenses correctly ',()=>{
    wrapper = shallow(<ExpensesListSummary expensesCount={2} expensesTotal={9434}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should display two expenses correctly ',()=>{
    wrapper = shallow(<ExpensesListSummary expensesCount={1} expensesTotal={9434}/>);
    expect(wrapper).toMatchSnapshot();
});

