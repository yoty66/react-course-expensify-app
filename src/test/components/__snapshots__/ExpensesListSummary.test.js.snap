// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display two expenses correctly  1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       
      expenses 
       totaling 
      <span>
        ₪94.34
      </span>
    </h1>
  </div>
  <div
    className="page-header__actions"
  >
    <Link
      className="button"
      to="/create"
    >
      Add Expense
    </Link>
  </div>
</div>
`;

exports[`should display two expenses correctly  2`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense 
       totaling 
      <span>
        ₪94.34
      </span>
    </h1>
  </div>
  <div
    className="page-header__actions"
  >
    <Link
      className="button"
      to="/create"
    >
      Add Expense
    </Link>
  </div>
</div>
`;
